src/dataworker/DataworkerUtils.ts

```diff
@@ -354,13 +355,17 @@ export function _getRefundLeaves(
  * @param chainId chain to check for WETH and ETH addresses
  * @returns WETH and ETH addresses.
  */
-function getWethAndEth(chainId: number): string[] {
+function getNativeTokens(chainId: number): string[] {
+  const nativeTokenSymbol = getNativeTokenSymbol(chainId);
   // Can't use TOKEN_SYMBOLS_MAP for ETH because it duplicates the WETH addresses, which is not correct for this use case.
-  const wethAndEth = [TOKEN_SYMBOLS_MAP.WETH.addresses[chainId], CONTRACT_ADDRESSES[chainId].eth.address];
-  if (wethAndEth.some((tokenAddress) => !isDefined(tokenAddress))) {
-    throw new Error(`WETH or ETH address not defined for chain ${chainId}`);
+  const nativeTokens = [
+    TOKEN_SYMBOLS_MAP[`W${nativeTokenSymbol}`].addresses[chainId],
```

Comment:
We might not make it to the asswert below if the token symbol doesn't exist at all.

```suggestion
    TOKEN_SYMBOLS_MAP[`W${nativeTokenSymbol}`]?.addresses[chainId],
```